### PR TITLE
Draftofmethods

### PR DESCRIPTION
**Pull request by barbara79**
1. code and add tests that cover the created code. Your code should be warning-free.
   * [ ] tests existing
1. stick to PSR-2 and and don't reformat existing code.
   * [ ] code style check succesful

### CODE REVIEW

File: src/Core/Model/Common/AssetDraft.php [hunk 1]
```diff
@@ -37,4 +37,36 @@ public function fieldDefinitions()
...
         ];
     }
+
+    /**
+     * @param AssetSourceCollection $sources
+     * @param LocalizedString $name
+     * @param Context|callable $context
+     * @return AssetDraft
+     */
+    public static function ofSourcesAndName(AssetSourceCollection $sources, LocalizedString $name, $context = null)
```
Comment (jenschude) [inline src/Core/Model/Common/AssetDraft.php]: maybe better change the order of parameters to $name and $sources
```suggestion
    public static function ofNameAndSources(LocalizedString $name, AssetSourceCollection $sources, $context = null)
```

File: src/Core/Model/Order/LineItemImportDraft.php [hunk 2]
```diff
@@ -60,4 +61,33 @@ public function fieldDefinitions()
...
+
+    /**
+     * @param LocalizedString $name
+     * @param Price $price
+     * @param ProductVariantImportDraft $variant
+     * @param int $quantity
+     * @param Context|callable $context
+     * @return LineItemImportDraft
+     */
+    public static function ofNamePriceVariantAndQuantity($name, $price, $variant, $quantity, $context = null)
```
Comment (jenschude) [inline src/Core/Model/Order/LineItemImportDraft.php]:
```suggestion
    public static function ofNamePriceVariantAndQuantity($name, $price, ProductVariantImportDraft $variant, $quantity, $context = null)
```

File: src/Core/Model/Order/ShippingInfoImportDraft.php [hunk 3]
```diff
@@ -54,4 +55,26 @@ public function fieldDefinitions()
...
+
+    /**
+     * @param string $shippingMethodName
+     * @param Money $price
+     * @param ShippingRate $shippingRate
+     * @param string $shippingMethodState
+     * @param Context|callable $context
+     * @return ShippingInfoImportDraft
+     */
+    public static function ofNamePriceRateDeliveryAndState(
```
Comment (jenschude) [inline src/Core/Model/Order/ShippingInfoImportDraft.php]: The name is mismatching the parameters

File: src/Core/Model/Product/ProductVariantDraft.php [hunk 4]
```diff
@@ -41,4 +43,35 @@ public function fieldDefinitions()
...
+        return static::of($context)->setSku($sku);
+    }
+
+    /**
+     * @param string $sku
+     * @param PriceDraftCollection $prices
+     * @param Context|callable $context
+     * @return ProductVariantDraft
+     */
+    public static function ofSkuAndPrices($sku, $prices, $context = null)
```
Comment (jenschude) [inline src/Core/Model/Product/ProductVariantDraft.php]:
```suggestion
    public static function ofSkuAndPrices($sku, PriceDraftCollection $prices, $context = null)
```

File: src/Core/Model/Product/ProductVariantDraft.php [hunk 5]
```diff
@@ -41,4 +43,35 @@ public function fieldDefinitions()
...
+    {
+        return static::of($context)->setSku($sku)->setPrices($prices);
+    }
+
+    /**
+     * @param PriceDraftCollection $prices
+     * @param Context|callable $context
+     * @return ProductVariantDraft
+     */
+    public static function ofPrices($prices, $context = null)
```
Comment (jenschude) [inline src/Core/Model/Product/ProductVariantDraft.php]:
```suggestion
    public static function ofPrices(PriceDraftCollection $prices, $context = null)
```

File: src/Core/Model/ShoppingList/TextLineItemDraft.php [hunk 6]
```diff
@@ -40,4 +41,14 @@ public function fieldDefinitions()
             'custom' => [static::TYPE => CustomFieldObjectDraft::class],
         ];
     }
+
+    /**
+     * @param LocalizedString $name
+     * @param Context|null $context
+     * @return TextLineItemDraft
+     */
+    public static function ofName($name, $context = null)
```
Comment (jenschude) [inline src/Core/Model/ShoppingList/TextLineItemDraft.php]:
```suggestion
    public static function ofName(LocalizedString $name, $context = null)
```

File: tests/integration/Me/MeOrderRequestTest.php [hunk 7]
```diff
@@ -39,28 +39,21 @@ class MeOrderRequestTest extends ApiTestCase
...
-                        LineItemDraft::of()
-                            ->setProductId($this->getProduct()->getId())
-                            ->setVariantId(1)
-                            ->setQuantity(1)
+                        LineItemDraft::ofProductIdVariantIdAndQuantity($this->getProduct()->getId(), 1, 1)
                     )
-            )
-            ->setShippingMethod($this->getShippingMethod()->getReference())
-        ;
+            );
```
Comment (jenschude) [inline tests/integration/Me/MeOrderRequestTest.php]: This doesn't need to be changed

File: tests/integration/Me/MeOrderRequestTest.php [hunk 8]
```diff
@@ -70,28 +63,21 @@ protected function getCartDraft()
...
-                        MyLineItemDraft::of()
-                            ->setProductId($this->getProduct()->getId())
-                            ->setVariantId(1)
-                            ->setQuantity(1)
+                        MyLineItemDraft::ofProductIdVariantIdAndQuantity($this->getProduct()->getId(), 1, 1)
                     )
-            )
-            ->setShippingMethod($this->getShippingMethod()->getReference())
-        ;
+            );
```
Comment (jenschude) [inline tests/integration/Me/MeOrderRequestTest.php]: This also doesn't need to be changed

File: tests/integration/Me/MeOrderRequestTest.php (hunk 7 above)
Comment (barbara79) [reply tests/integration/Me/MeOrderRequestTest.php]: done but why ;)

File: tests/integration/Me/MeOrderRequestTest.php (hunk 8 above)
Comment (barbara79) [reply tests/integration/Me/MeOrderRequestTest.php]: done but why ;)

File: tests/integration/OrderEdit/OrderEditUpdateRequestTest.php [hunk 9]
```diff
@@ -276,9 +276,10 @@ public function stagedActionsProvider()
             StagedOrderAddPaymentAction::class => [function() { return StagedOrderAddPaymentAction::of()->setPayment($this->getPayment()->getReference()); }],
             StagedOrderRemovePaymentAction::class => [function() { return StagedOrderRemovePaymentAction::of()->setPayment($this->getPayment()->getReference()); }],
             StagedOrderSetShippingMethodTaxAmountAction::class => [function() { return StagedOrderSetShippingMethodTaxAmountAction::of()->setExternalTaxAmount(
-                ExternalTaxAmountDraft::of()
-                    ->setTotalGross(Money::ofCurrencyAndAmount('EUR', 100))
-                    ->setTaxRate(ExternalTaxRateDraft::ofNameCountryAndAmount($this->getTestRun() . '-name','DE',100.00))
+                ExternalTaxAmountDraft::ofTotalGrossAndTaxRate(
+                    Money::ofCurrencyAndAmount('EUR', 100),
+                    ExternalTaxRateDraft::ofNameCountryAndAmount($this->getTestRun() . '-name','DE',100.00)
+                )
```
Comment (jenschude) [inline tests/integration/OrderEdit/OrderEditUpdateRequestTest.php]: Test

**Pull request merged**